CRM/Pledge/BAO/Pledge.php

```diff
@@ -1202,4 +1202,36 @@ protected static function getTransactionalStatus() {
     return array_diff(array_flip($paymentStatus), self::getNonTransactionalStatus());
   }
 
+  /**
+   * Create recur record for pledge.
+   *
+   */
+  public static function createRecurRecord($pledge, $params, $form) {
+    $recurParams = array('contact_id' => $pledge->contact_id);
+    $recurParams['amount'] = $pledge->amount;
+    $recurParams['auto_renew'] = TRUE;
+    $recurParams['frequency_unit'] = $pledge->frequency_unit;
+    $recurParams['frequency_interval'] = $pledge->frequency_interval;
+    $recurParams['installments'] = $pledge->installments;
+    $recurParams['financial_type_id'] = $pledge->financial_type_id;
+    $recurParams['currency'] = $pledge->currency;
+
+    if (!empty($pledge->start_date)) {
+      $recurParams['start_date'] = $pledge->start_date;
+    }
+    $recurParams['invoice_id'] = CRM_Utils_Array::value('invoiceID', $params);
+    $recurParams['contribution_status_id'] = "Pending";
+    $recurParams['payment_processor_id'] = CRM_Utils_Array::value('payment_processor_id', $params);
+    $recurParams['is_email_receipt'] = CRM_Utils_Array::value('is_email_receipt', $params);
+    $recurParams['trxn_id'] = CRM_Utils_Array::value('trxn_id', $params, $params['invoiceID']);
+    $recurParams['financial_type_id'] = CRM_Utils_Array::value('financial_type_id', $params);
+
+    if (!empty($form->_values['is_monetary'])) {
```

Comment:
Hi @eileenmcnaughton,

I basically just followed what core was doing to create a recurring contribution. https://github.com/civicrm/civicrm-core/blob/master/CRM/Contribute/Form/Contribution/Confirm.php#L1087
Is payment_instrument_id intended to be set to 1 here?